### DISCUSSION TITLE
How to customize enum mapping?

### DISCUSSION THREAD

**Discussion by maciejkopecpl** [graphql-java-kickstart/graphql-java-tools#426] [Q&A] [answered]
Hey, 

Is there any way to customize how enum is mapped/serialized? I have a following Java enum:
```java
public enum CustomEnum {
  VALUE_1("value1"),
  VALUE_2("value2")
}
```

schema:
```
type MyValue{
    id: ID
    type: CustomEnum 
}
```

And when I run a query I get the following results:
```
{
  "data": {
    "myValue": {
      "__typename": "MyValue",
      "type": "VALUE_1"
    }
  }
}
```

but I want to have the following:

```
{
  "data": {
    "myValue": {
      "__typename": "MyValue",
      "type": "value1"
    }
  }
}
```

Is there any way to customize how enum is mapped? 

Thanks for any help!

**Comment by priyaaggarwal24** [comment, marked as answer]
Value of Enum type cannot be customized as of current implementation.
To get the result that you're looking for, you can create a custom scalar that wraps Enum and uses the property of Enum that you want

**Comment by maciejkopecpl** [comment]
Yeah, that's what I did, however, it introduces duplication. It would be nice if it would be possible to configure such behavior.
